File: proxy/tests/script_engine/url_scheme.js
/// <reference path="../../script_libs/js/roxy.d.ts" />
/** @type {Extension} */
const url_scheme = {
  request(flow) {
    if (flow.request.url.protocol == "http") {
      flow.request.url.protocol = "https";
    }
  },
};
globalThis.extensions = [url_scheme];

File: proxy/tests/script_engine/url_scheme.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local url_scheme = {
	request = function(flow)
		if flow.request.url.protocol == "http" then
			flow.request.url.protocol = "https"
		end
	end,
}
Extensions = { url_scheme }
